wire/msgcfilter_test.go

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) 2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package wire
+
+import (
+	"bytes"
+	"io"
+	"reflect"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+)
+
+// TestCFilter tests the MsgCFilter API.
+func TestCFilter(t *testing.T) {
+	pver := ProtocolVersion
+
+	// Block 200,000 hash.
+	hashStr := "000000000000007a59f30586c1003752956a8b55e6f741fd5f24c800cd5e5e8c"
+	blockHash, err := chainhash.NewHashFromStr(hashStr)
+	if err != nil {
+		t.Fatalf("NewHashFromStr: %v", err)
+	}
+
+	// Arbitrary CF data.
+	data := make([]byte, 32)
+
+	// Ensure the command is expected value.
+	wantCmd := "cfilter"
+	msg := NewMsgCFilter(blockHash, GCSFilterExtended, data)
+	if cmd := msg.Command(); cmd != wantCmd {
+		t.Fatalf("NewMsgCFilter: wrong command - got %v want %v",
+			cmd, wantCmd)
+	}
+
+	// Ensure max payload is expected value for latest protocol version.
+	// Block hash 32 bytes + filter type 1 byte + CF data size (varInt) 5 bytes
+	// + max CF data size.
+	wantPayload := uint32(262182)
+	maxPayload := msg.MaxPayloadLength(pver)
+	if maxPayload != wantPayload {
+		t.Fatalf("MaxPayloadLength: wrong max payload length for "+
+			"protocol version %d - got %v, want %v", pver,
+			maxPayload, wantPayload)
+	}
+
+	// Ensure we get the same data out.
+	if !reflect.DeepEqual(&msg.BlockHash, blockHash) {
+		t.Fatalf("NewMsgCFilter: wrong block hash - got %v, want %v",
+			spew.Sdump(&msg.BlockHash), spew.Sdump(blockHash))
+	}
+	if msg.FilterType != GCSFilterExtended {
+		t.Fatalf("NewMsgCFilter: wrong filter type - got %v, want %v",
+			spew.Sdump(msg.FilterType), spew.Sdump(GCSFilterExtended))
+	}
+	if !reflect.DeepEqual(msg.Data, data) {
```

Comment:
Please use `if !bytes.Equal(msg.Data, data) {` instead for the reason previous described.